functional-tests/src/main/java/com/redhat/che/selenium/core/workspace/CheStarterWrapper.java

```diff
@@ -133,64 +163,104 @@ public void startWorkspace(String WorkspaceID, String name, String token) throws
     sendStartRequest(WorkspaceID, token);
   }
 
-  public void sendStartRequest(String id, String token) throws IOException {
-    OkHttpClient client = new OkHttpClient();
-    RequestBody body = RequestBody.create(null, new byte[0]);
-    StringBuilder sb = new StringBuilder("https://" + this.host);
-    sb.append("/api/workspace/");
-    sb.append(id);
-    sb.append("/runtime");
-    Builder requestBuilder = new Request.Builder().url(sb.toString());
-    requestBuilder.addHeader("Authorization", "Bearer " + token);
-    Request request = requestBuilder.post(body).build();
+  public void sendStartRequest(String id, String token) throws Exception {
+    String relativePath = "workspace/" + id + "/runtime";
+    String authorization = "Bearer " + token;
+    CheServerStartWorkspaceRequest body = new CheServerStartWorkspaceRequest();
+    HttpJsonResponse response = null;
     try {
-      Response response = client.newCall(request).execute();
-      if (response.isSuccessful()) {
-        LOG.info("Workspace was started. Waiting until workspace is running.");
-      }
-    } catch (IOException e) {
-      LOG.error("Workspace start failed : " + e.getMessage(), e);
+      response =
+          this.jsonRequestFactory
+              .fromUrl(testApiEndpointUrlProvider.get().toString().concat(relativePath))
+              .setAuthorizationHeader(authorization)
+              .setBody(body)
+              .setMethod(HttpMethod.POST)
+              .request();
+    } catch (ServerException e) {
+      LOG.error("Workspace failed to start because of a server error:" + e.getMessage(), e);
+      throw e;
+    } catch (ForbiddenException e) {
+      LOG.error("Workspace failed to start because of a forbidden request:" + e.getMessage(), e);
+      throw e;
+    } catch (BadRequestException e) {
+      LOG.error("Workspace failed to start because of a bad request:" + e.getMessage(), e);
+      throw e;
+    } catch (ConflictException e) {
+      LOG.error("Workspace fauled to start because of a conflicting request:" + e.getMessage(), e);
       throw e;
+    } catch (NotFoundException e) {
+      LOG.error("The requested endpoint returned 404:" + e.getMessage(), e);
+      throw e;
+    } catch (UnauthorizedException e) {
```

Comment:
Wouldn't it be nicer to catch `org.eclipse.che.api.core.ApiException`? I don't see much value in catching every exception on it's own just to LOG different message (it seems kind of redundant to me, as the exception already contains information about nature of the failure)

Reply:
will do.